### PR TITLE
0.5 fix

### PR DESCRIPTION
**Pull request by samuel-massinon**

### CODE REVIEW

**Comment by omus** [comment]
Note that `readstring` hasn't been backported to v0.4 so you'll want to use the [Compat.jl](https://github.com/JuliaLang/Compat.jl) package to maintain compatibility with 0.4 and 0.5.

**Comment by samuel-massinon** [comment]
@omus Can you do a code review?

File: src/FTPC.jl [hunk 1]
```diff
@@ -22,9 +22,9 @@ type RequestOptions
 
         if url == nothing
             if implicit
-                url = "ftps://" * String(hostname) * "/"
+                url = "ftps://" * AbstractString(hostname) * "/"
             else
-                url = "ftp://"* String(hostname) * "/"
+                url = "ftp://"* AbstractString(hostname) * "/"
```
Comment (omus) [inline src/FTPC.jl]: This is strange... you probably want to do `string(hostname)` here.

File: test/runtests.jl [hunk 2]
```diff
@@ -16,17 +17,17 @@ function stop_server()
...
 end
 
-function set_file(name::String, content::String)
+function set_file(name::AbstractString, content::AbstractString)
     result = jcall(MockFTPServerJulia, "setFile", jboolean, (JString, JString), name, content)
     @assert result == 1 "set_file failed"
 end
 
-function set_command_response(request::String, code::Int64, reponse::String)
+function set_command_response(request::AbstractString, code::Int64, reponse::AbstractString)
```
Comment (omus) [inline test/runtests.jl]: Probably could also switch the `Int64` here to `Integer`.

File: test/test_non_blocking.jl [hunk 3]
```diff
@@ -52,7 +52,7 @@ context("Changed directory and get file") do
     rcall = ftp_connect(options)
     ctxt, resp = fetch(rcall)
     @fact resp.code --> 226
-    text = readall(resp.body)
+    text = readstring(resp.body)
```
Comment (omus) [inline test/test_non_blocking.jl]: Looks like you're missing an `@compat` here.

**Comment by samuel-massinon** [comment]
@omus Is it good to :shipit:?

**Comment by omus** [comment]
LGTM. My only gripe is with the travis.yml changes but I think that is a necessary evil.

**Comment by samuel-massinon** [comment]
@omus: Yeah, it's not pretty. Not sure when JavaCall.jl will fix that issue but I'll keep an eye on it https://github.com/aviks/JavaCall.jl/pull/30.

**Pull request merged**